src/python/pants/backend/python/subsystems/lambdex.py

```diff
@@ -22,7 +22,7 @@ class Lambdex(PythonToolBase):
     default_main = ConsoleScript("lambdex")
 
     register_interpreter_constraints = True
-    default_interpreter_constraints = ["CPython>=3.7,<3.10"]
+    default_interpreter_constraints = ["CPython<4,>=3.7"]
```

Comment:
I think it would make sense to keep the defaults to line up with what we say we support in `PythonGoogleCloudFunctionRuntimes`.

Reply:
Yes. Since you need to add an enum value anyhow, the apparent freedom of the looser constraint buys you ~nothing. Perhaps add a comment to the enum that points to the subsystem though reminding that it needs to have it's IC widened. The whole thing is definitely awkward since Lambdex serves two masters, but, at least for now, Google is ahead of AWS and dictates the upper bound.

Reply:
Does it mean just to use a little bit stricter interpreter constraints? Like this `default_interpreter_constraints = ["CPython>=3.7,<=3.11"]` since `python311` is the newest supported python version by Google?

Reply:
Yes, but `>=3.7,<3.12` is what you want. Your suggestion only works for 3.11.0 and not 3.11.1 and higher.

Reply:
You are right. Updated in e87e6db060e62474b9383dad8636f97cfa2ef88a